core/src/main/kotlin/com/strumenta/kolasu/ids/SourceIdProvider.kt

```diff
@@ -24,15 +25,15 @@ abstract class AbstractSourceIdProvider : SourceIdProvider {
 
 class SimpleSourceIdProvider(var acceptNullSource: Boolean = false) : AbstractSourceIdProvider() {
     override fun sourceId(source: Source?): String {
-        if (source is IDLogic) {
-            return source!!.calculatedID(RootCoordinates)
+        if (source is SemanticNodeIDProvider) {
+            return source!!.semanticID(TODO())
```

Comment:
Currently I don't see any Source implementation which is also a SemanticNodeIDProvider, so what's the point of this branch?

Reply:
Yes, that wrong. I am removing it